include/roboptim/core/function.hh

```diff
@@ -357,6 +369,33 @@ namespace roboptim
       return boost::get<2> (interval);
     }
 
+    /// \brief Get the type-checking flag
+    virtual unsigned long flag()
+    {
+      return flag_;
+    }
+
+    /// \brief Fonction cast as.
+    ///
+    /// \tparam ExpectedType type we want to cast the function into
+    template <class ExpectedType>
+    ExpectedType* castInto()
+    {
+      if (asType<ExpectedType>())
+        return reinterpret_cast<ExpectedType*>(this);
```

Comment:
As the asType method does the type checking, both are theoretically the same. I avoided `static_cast` because I thought there might be a little overhead (even only compile-time), but it seems not really to be significant (RobOptim is no real-time/low-level stuff anyway, so no need to go that far in time execution optimization).
I understand very much `reinterpret_cast` are "cast non grata". I will change it :wink: